Add interim fix for typename error when linking drafts

uoa-noel opened the pull request:
This PR adds an interim fix for typename error when content links to draft content, by handling the server returning null content. Still need to investigate cer-graphql issues that caused this typename error.

rosemcc commented:
Nice, it works! This might be a dumb question, but do we just need to do that just on subhubs and cards, or do we need something similar on articles, equipment etc pages too?

uoa-noel commented:
Thanks for picking that up @rosemcc . The different content type components all use the `cards` and `large-cards` components, and I've changed the card components to handle nulls. However, the content type components will still add a container with its own styling when there is null content. So I added code for removing null content in the content type components as well. In addition, the rich media body component (`BodyMediaComponent` - for displaying body content that have embedded content, like inline images, files or links) is also changed to handle null content.

In the future we should code more defensively against null content as GraphQL makes fields nullable by default, and spec requires servers to return a null if a resolver runs into an error. (https://hasura.io/blog/graphql-nulls-cheatsheet/)

Also, we should look more into and raise with Contentful the inconsistent way it handles draft reference content of one type and of multiple types.

I've also run component and e2e tests locally to ensure they pass.

Unless there are any other comments, I will merge and close this PR!

rosemcc commented:
Nice work Noel!! Happy for this to be merged :)
